day34/validateIP.py

```diff
@@ -0,0 +1,74 @@
+def validateIP(ip):
+    ipv4 = ip.split(".")
+    ipv6 = ip.split(":")
+    if len(ipv4) == 4 and validateIPv4(ipv4):
+        return "IPv4"
+    elif len(ipv6) == 8 and validateIPv6(ipv6):
+        return "IPv6"
+    else:
+        return "Neither"
+
+def validateIPv4(segments):
+    digits = set("1234567890")
+    for s in segments:
+        # segment too long or too short
+        if len(s) > 3 or len(s) == 0:
+            return False
+
+        # segment has leading 0
+        if len(s) > 1 and s[0] == "0":
```

Comment:
`s.startswith('0')` makes a bit more readable here, and you can remove the comment